src/OpenTelemetry/Metrics/Exemplar/ReadOnlyExemplarCollection.cs

```diff
@@ -50,24 +51,39 @@ public Enumerator GetEnumerator()
 
     internal ReadOnlyExemplarCollection Copy()
     {
-        var exemplarCopies = new Exemplar[this.exemplars.Length];
+        var maximumCount = this.MaximumCount;
 
-        int i = 0;
-        foreach (ref readonly var exemplar in this)
+        if (maximumCount > 0)
         {
-            exemplar.Copy(ref exemplarCopies[i++]);
+            var exemplarCopies = new Exemplar[maximumCount];
+
+            int i = 0;
+            foreach (ref readonly var exemplar in this)
+            {
+                if (exemplar.IsUpdated())
+                {
+                    exemplar.Copy(ref exemplarCopies[i++]);
+                }
+            }
+
+            return new ReadOnlyExemplarCollection(exemplarCopies);
         }
 
-        return new ReadOnlyExemplarCollection(exemplarCopies);
+        return Empty;
     }
 
     internal IReadOnlyList<Exemplar> ToReadOnlyList()
     {
         var list = new List<Exemplar>(this.MaximumCount);
 
-        foreach (var item in this)
+        foreach (var exemplar in this)
         {
-            list.Add(item);
+            if (exemplar.IsUpdated())
+            {
+                var newExemplar = default(Exemplar);
```

Comment:
Why do we create a copy and add it to the list? Was there an issue with directly adding the exemplar here?

Reply:
Sort of, but not really 😄 This is just used by tests which we know are using in-memory exporter which first calls the `Copy` method above. I just reverted the change and replaced it with a comment to explain the situation.